### PR TITLE
Fix encoding issues + IDNA support

### PR DESCRIPTION
**Pull request by MuxZeroNet**
Please test it on your own operating system and on your own locale.

### CODE REVIEW

File: src/Ui/UiRequest.py [hunk 1]
```diff
@@ -40,11 +41,17 @@ def __init__(self, server, get, env, start_response):
...
+
     # Call the request handler function base on path
     def route(self, path):
         if config.ui_restrict and self.env['REMOTE_ADDR'] not in config.ui_restrict:  # Restict Ui access by ip
             return self.error403(details=False)
 
+        self.log.debug("-- path repr: " + repr(path))
+        path = path.decode('utf-8')
+        self.log.debug("-- path decoded repr: " + repr(path))
+
```
Comment (MuxZeroNet) [inline src/Ui/UiRequest.py]: If you are annoyed by encoding issues of Python 2, try using `repr(...)` on strings when printing out debug log.

**Comment by MuxZeroNet** [comment]
Solves issues #131 and #298 but has homograph attacks. It may be a good idea to show a warning to show the encoded domain name, the domain public key, and whether known homographs are found in the domain name.

**Comment by MuxZeroNet** [comment]
I decided not to make ZeroNet support IDNA domain names simply because it will make browsing IDNA sites more inconvenient.

Suppose someone registered for a fancy domain name `☉net.bit` (the Sun symbol), and a bad guy registered for a domain name `⊙net.bit` (the circled dot operator) for phishing.
```python
>>> u'☉' == u'⊙'
False
>>> print repr(u'☉'),repr(u'⊙')
u'\u2609' u'\u2299'
```

Most fonts are not optimized for differentiating these homographs, so we want ZeroNet to display them in puny coded form.
```python
>>> u'⊙net.bit'.encode('idna'), u'☉net.bit'.encode('idna')
('xn--net-vr2a.bit', 'xn--net-gn5a.bit')
```
To not to get to the phishing site, one has to memorize the full puny coded form of the innocent site's domain name. In this case, one will choose to use public keys to access these sites instead.

I will create a separate pull request only to fix encoding issues.

**Comment by HelloZeroNet** [comment]
Yeah, I'm also uncertain about utf8 domain names, no one can tell the difference between Оnet, ⵔnet, Onet, Οnet or Onet (All uses different "0" character)

**Comment by MuxZeroNet** [comment]
Closed. IDNA domains will not be supported due to security concerns. Discussion about string encoding bugs has been moved here: #765

**Pull request closed without merging**